system-x/services/flink/src/main/java/software/tnb/flink/resource/openshift/OpenshiftFlink.java

```diff
@@ -0,0 +1,168 @@
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.readiness.Readiness;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+
+@AutoService(Flink.class)
+public class OpenshiftFlink extends Flink implements OpenshiftDeployable, WithExternalHostname, WithOperatorHub, WithCustomResource, WithDockerImage,
+    WithName {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpenshiftFlink.class);
+
+    private Route apiRoute;
+
+    @Override
+    public String operatorChannel() {
+        return "alpha";
+    }
+
+    @Override
+    public String operatorName() {
+        return "flink-kubernetes-operator";
+    }
+
+    @Override
+    public String operatorCatalog() {
+        return "community-operators";
+    }
+
+    @Override
+    public void undeploy() {
+        LOG.info("Undeploying Flink resources");
+        OpenshiftClient.get().genericKubernetesResources(apiVersion(), kind()).delete();
+        WaitUtils.waitFor(() -> servicePod() == null, "Waiting until the pod is removed");
+    }
+
+    @Override
+    public void openResources() {
+        apiRoute = OpenshiftClient.get().routes().createOrReplace(new RouteBuilder()
+            .editOrNewMetadata()
+            .withName(name())
+            .endMetadata()
+            .editOrNewSpec()
+            .withNewTo().withKind("Service").withName(host()).withWeight(100)
+            .endTo()
+            .withNewPort().withTargetPort(new IntOrString("rest")).endPort()
+            .endSpec()
+            .build());
+
+        WaitUtils.waitFor(() -> HTTPUtils.getInstance().get(externalHostname(), false).isSuccessful()
+            , 5, 5000L
+            , "Waiting until the Flink API route is ready");
+    }
+
+    @Override
+    public void closeResources() {
+        if (apiRoute != null) {
+            OpenshiftClient.get().routes().resource(apiRoute).delete();
+        }
+    }
+
+    @Override
+    public void create() {
+        LOG.debug("Creating Flink subscription");
+
+        // Create subscription if needed
+        createSubscription();
+
+        WaitUtils.waitFor(() -> OpenshiftClient.get().pods().withLabel("app.kubernetes.io/name", "flink-kubernetes-operator")
+            .list().getItems().size() > 0 && OpenshiftClient.get().pods().withLabel("app.kubernetes.io/name", "flink-kubernetes-operator").list()
+            .getItems().stream().allMatch(Readiness::isPodReady), "Wait for operator PODs are ready");
+
+        OpenshiftClient.get().genericKubernetesResources(apiVersion(), kind()).resource(customResource()).create();
+    }
+
+    @Override
+    public boolean isDeployed() {
+        List<Pod> pods = OpenshiftClient.get().pods().withLabel("app.kubernetes.io/name", "flink-kubernetes-operator").list().getItems();
+        return pods.size() == 1 && pods.stream().allMatch(Readiness::isPodReady)
+            && servicePods().size() == 2 && isReady();
```

Comment:
`isDeployed` should not use `isReady` method

Reply:
I think count service pod is enough, thanks